### PR TITLE
[TT-12698] Add linter to gateway to check for named scope conflicts

### PR DESCRIPTION
**Pull request by titpetric**
### **PR Type**
Enhancement, Tests

https://tyktech.atlassian.net/browse/TT-12698

___

### **Description**
- Renamed imports for internal GraphQL and temporal model packages to avoid named scope conflicts.
- Updated usage of renamed imports in `gateway/handler_success.go` and `gateway/redis_signals.go`.
- Added a new linter task `lint:gateway` to the `Taskfile.yml` to run `go-fsck lint` in the `gateway` directory.
- Updated `lint:all` task in `Taskfile.yml` to include `lint:gateway` as a dependency.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>handler_success.go</strong><dd><code>Rename internal GraphQL package import and update usage</code>&nbsp; &nbsp; </dd></summary>
<hr>

gateway/handler_success.go

<li>Renamed import for internal GraphQL package to <code>graphqlInternal</code>.<br> <li> Updated usage of <code>graphql</code> to <code>graphqlInternal</code> for <br><code>NewGraphStatsExtractor</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6409/files#diff-45135957493eca37f2e3e9a81079577777133c53b27cf95ea2ff0329c05bd006">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>redis_signals.go</strong><dd><code>Rename temporal model package import and update usage</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/redis_signals.go

<li>Renamed import for temporal model package to <code>temporalModel</code>.<br> <li> Updated usage of <code>model</code> to <code>temporalModel</code> for <code>Message</code> and <br><code>MessageTypeMessage</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6409/files#diff-18cb136722c238e19b02741a85510dfd464343f85365482f0873aa60a37718af">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Taskfile.yml</strong><dd><code>Add linter task for gateway and update lint:all dependencies</code></dd></summary>
<hr>

Taskfile.yml

<li>Added new task <code>lint:gateway</code> to run <code>go-fsck lint</code> in the <code>gateway</code> <br>directory.<br> <li> Updated <code>lint:all</code> task to include <code>lint:gateway</code> as a dependency.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6409/files#diff-cd2d359855d0301ce190f1ec3b4c572ea690c83747f6df61c9340720e3d2425e">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: gateway/api_definition_test.go [hunk 1]
```diff
@@ -15,15 +15,13 @@ import (
...
 
-	"github.com/TykTechnologies/tyk/apidef/oas"
-
-	"github.com/TykTechnologies/storage/persistent/model"
-	"github.com/TykTechnologies/tyk/config"
-	"github.com/TykTechnologies/tyk/rpc"
-
 	"github.com/stretchr/testify/assert"
 
+	persistentModel "github.com/TykTechnologies/storage/persistent/model"
```
Comment (jeffy-mathew) [inline gateway/api_definition_test.go:20]: this could be all in lowercase letters, right?

Comment (titpetric) [reply gateway/api_definition_test.go:20]: we don't have a convention other than the established one, it's mostly this :)

File: apidef/adapter/gqlengineadapter/adapter_udg.go [hunk 2]
```diff
@@ -65,9 +65,9 @@ func (u *UniversalDataGraph) engineConfigV2FieldConfigs() (planFieldConfigs plan
 	return planFieldConfigs
 }
 
-func (u *UniversalDataGraph) engineConfigV2DataSources() (planDataSources []plan.DataSourceConfiguration, err error) {
+func (u *UniversalDataGraph) engineConfigV2datasources() (plandatasources []plan.DataSourceConfiguration, err error) {
```
Comment (jeffy-mathew) [inline apidef/adapter/gqlengineadapter/adapter_udg.go:68]: this is renaming variables as well??

**Comment by titpetric** [comment]
Sonarcloud reports existing issues and coverage as refactoring touched a lot of files; cannot address as the scope is too large/preexisting, so we're making a merge exception here due to the nature of the change (type safe import replacements).

File: internal/graphengine/engine_v3_reverse_proxy.go [hunk 3]
```diff
@@ -9,10 +9,10 @@ import (
 
 	"github.com/jensneuse/abstractlogger"
 
-	"github.com/TykTechnologies/graphql-go-tools/v2/pkg/engine/datasource/httpclient"
-	"github.com/TykTechnologies/graphql-go-tools/v2/pkg/graphql"
-	"github.com/TykTechnologies/graphql-go-tools/v2/pkg/subscription"
-	gqlwebsocketV2 "github.com/TykTechnologies/graphql-go-tools/v2/pkg/subscription/websocket"
+	httpclientv2 "github.com/TykTechnologies/graphql-go-tools/v2/pkg/engine/datasource/httpclient"
```
Comment (jeffy-mathew) [inline internal/graphengine/engine_v3_reverse_proxy.go:12]: these packages are already on v2, do we need v2 suffix in the package name alias?

Comment (titpetric) [reply internal/graphengine/engine_v3_reverse_proxy.go:12]: because of conflicts in package scope, there's also a v1

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**